websms.py

```diff
@@ -0,0 +1,442 @@
+            'messageContent': message_content,
+            'test': test,
+        }
+
+        sender_address = sender_address or self.sender_address
+        if sender_address:
+            data['senderAddress'] = sender_address
+
+        sender_address_type = sender_address_type or self.sender_address_type
+        if sender_address_type:
+            data['senderAddressType'] = sender_address_type
+
+        max_sms_per_message = max_sms_per_message or self.max_sms_per_message
+        if max_sms_per_message:
+            data['maxSmsPerMessage'] = str(max_sms_per_message)
+
+        if timeout is None:
+            timeout = self.timeout
+
+        return self._request(
+            url=self.post_url,
+            data=data,
+            auth=auth,
+            timeout=timeout,
+            resource_name='Post SMS',
+            fail_silently=fail_silently,
+        )
+
+    @staticmethod
+    def _request(
+        url: str,
+        data: dict,
+        auth: Tuple[Optional[str], Optional[str]],
+        timeout: float,
+        resource_name: str,
+        fail_silently: bool,
+    ) -> Optional[SMSSendResponse]:
+        """Sends a request to Service API, construct SMSSendResponse object from
+        response.
+
+        :param url: destination's URL address
+        :param data: POST data
+        :param auth: HTTP Basic Auth data
+        :param timeout: network timeout in seconds
+        :param resource_name: destination's verbose name (for logging)
+        :param fail_silently: do not raise exceptions
+
+        :raise RequestError: remote API responded with error message or
+            HTTP error status
+        :raise CommunicationError: unable to communicate with remote API
+
+        :return: `SMSSendResponse` object or `None`
+        """
+        try:
+            response = requests.post(
+                url=url, json=data, auth=auth, timeout=timeout
+            )
+            logger.debug(
+                '{} response: {} {}'.format(
+                    resource_name, response.status_code, response.reason
+                )
+            )
+            response.raise_for_status()
+        except requests.RequestException as error:
+            if isinstance(error, requests.HTTPError):
+                exception_class = RequestError
+            else:
+                exception_class = CommunicationError
+            exception = exception_class(original_exception=error)
+            logger.error(exception)
+
+            sms_response = None
+            if not fail_silently:
+                raise exception
+        else:
+            sms_response = SMSSendResponse(response.text)
+            if not sms_response.transfer_id:
+                exception = RequestError(sms_response.status_message)
+                logger.error(exception)
+
+                sms_response = None
+                if not fail_silently:
+                    raise exception
+
+        return sms_response
+
+
+default_service = SMSService()
```

Comment:
Wonder if this is global value - then maybe better is uppercase?

Reply:
IMO this is fine in lower case, as it's an object, and the user can call its methods or modify its state. Upper case constants seem more suitable for basic Python data structures.

Reply:
Then let's leave it :)